frameworks/fmi2api/src/main/java/org/intocps/maestro/framework/fmi2/api/mabl/scoping/ScopeFmi2Api.java

```diff
@@ -219,6 +216,92 @@ protected StringVariableFmi2Api store(Supplier<String> nameProvider, String valu
+     * @param mdArray non-jagged multidimensional array.
+     * @param <V> data type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArray(String identifyingName, V[] mdArray) {
+        List<Integer> arrayShape = new ArrayList<>();
+        PType type;
+        V[] subArr = mdArray;
+        while(subArr.getClass().getComponentType().isArray()){
+            arrayShape.add(subArr.length);
+            subArr = (V[])subArr[0];
+        }
+        arrayShape.add(subArr.length);
+
+        if (subArr instanceof Double[]) {
+            type = newARealNumericPrimitiveType();
+        } else if (subArr instanceof Integer[]) {
+            type = newAIntNumericPrimitiveType();
+        } else if (subArr instanceof Boolean[]) {
+            type = newABoleanPrimitiveType();
+        } else if (subArr instanceof String[]) {
+            type = newAStringPrimitiveType();
+        } else { throw new IllegalArgumentException();  }
+
+        PStm arrayVariableStm = newALocalVariableStm(
+                newAVariableDeclarationMultiDimensionalArray(newAIdentifier(identifyingName), type, arrayShape));
+
+        add(arrayVariableStm);
+
+        return createMDArrayRecursively(mdArray, arrayVariableStm,
+                newAIdentifierStateDesignator(newAIdentifier(identifyingName)), newAIdentifierExp(identifyingName));
+    }
+
+    /**
+     * @param array          multi dimensional array
+     * @param declaringStm   declaring statement of the outer array
+     * @param <V>            generic type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArrayRecursively(V[] array, PStm declaringStm, PStateDesignatorBase stateDesignator, PExpBase indexExp) {
+
+        if (array.getClass().getComponentType().isArray()) {
+            List<VariableFmi2Api> arrays = new ArrayList<>();
+            for (int i = 0; i < array.length; i++) {
+                arrays.add(createMDArrayRecursively((V[]) array[i], declaringStm,
+                        newAArayStateDesignator(stateDesignator, newAIntLiteralExp(i)), newAArrayIndexExp(indexExp,
+                                List.of(newAIntLiteralExp(i)))));
+            }
+            return new ArrayVariableFmi2Api(declaringStm, arrays.get(0).getType(), this, builder.getDynamicScope(),
+                    stateDesignator, indexExp.clone(), arrays);
+        }
+
+        List<VariableFmi2Api<V>> variables = new ArrayList<>();
+        for (int i = 0; i < array.length; i++) {
+            PType type = null;
+            Fmi2Builder.ExpressionValue value = null;
+
+            if (array instanceof Double[]) {
+                type = newARealNumericPrimitiveType();
+                value = new DoubleExpressionValue((Double) array[i]);
+            } else if (array instanceof Integer[]) {
+                type = newAIntNumericPrimitiveType();
```

Comment:
probably also need to include long

Reply:
Yes. I take it that it would map to AUIntNumericPrimitiveType?

Reply:
yes